resources/chromium/nfc-mock.js

```diff
@@ -72,6 +76,14 @@ function compareNDEFRecords(providedRecord, receivedRecord) {
 
   assert_not_equals(providedRecord.recordType, 'empty');
 
+  if (providedRecord.recordType == 'text') {
+    assert_equals(
+        providedRecord.encoding == null? 'utf-8': providedRecord.encoding,
+        receivedRecord.encoding);
+    assert_equals(providedRecord.lang == null? 'en': providedRecord.lang,
+                  receivedRecord.lang);
+  }
```

Comment:
For other record types, `encoding` and `lang` in |receivedRecord| should be always null, and we do not care about what values they are in |providedRecord| because the `NDEFRecord` creation algorithm does not force them to be null and just ignores them.